command/agent/command.go

```diff
@@ -77,6 +77,17 @@ func (c *Command) readConfig() *Config {
 	flags.Usage = func() { c.Ui.Error(c.Help()) }
 
 	// Role options
+
+	// Dev mode flags: the stdlib flag package doesn't allow non-bool flags
+	// to be passed an empty value. Although we can workaround this by setting
+	// IsBoolVar on the Value implementation, this results in unclear error
+	// messages. So we manipulate the args directly before passing into the
+	// Parse method.
+	for argIndex, arg := range c.args {
+		if arg == "-dev" || arg == "--dev" {
+			c.args[argIndex] = "-dev=true"
```

Comment:
Don't non-bool flags use their default value? Wouldn't a `-dev` definition like this work?

```go
devMode := ""
StringVar(&p, "dev", "default", "set the dev mode")
```
https://golang.org/pkg/flag/#StringVar

Reply:
Right, if the flag is omitted it'll be assigned the default value! But unfortunately that results in the bare `-dev` flag raising an error `flag needs an argument: -dev`.

So if we use a string flag, `-dev=connect` works, `-dev=default` works, and no dev flag at all works, but `-dev` breaks. Maybe there's a better flag parsing library that does a nicer job of this and/or exposes more of its guts for us to twiddle, but I wasn't sure we wanted to take on that level of investment for this fix?

See https://play.golang.org/p/jR2NI25892j for a minimal repro.